### DISCUSSION TITLE
Help decoding YUV420

### DISCUSSION THREAD

**Discussion by tgmichel** [ralfbiedert/openh264-rs#39] [General]
Hey! I'm trying to decode a raspberry pi captured bitstream which happens to be in `YUV420` format.

After [decoding](https://github.com/ralfbiedert/openh264-rs/blob/5fde2d014b91079c0d5fd0899dae06d6a62975a1/openh264/src/decoder.rs#L177) the Y, U and V components lengths are:

```
y_with_stride().len(): 2142720
u_with_stride().len(): 535680
v_with_stride(): 535680
```

`YUV420` byte array should have a `1920 * 1080 * 3 / 2 = 3110400` size, but with the decoded components I'd end up with a `2142720 + 535680 + 535680 = 3214080` length array, and at that point I'm lost at how to build a valid decoded byte array.

Could you please help me understand where the discrepancy is coming from? Thank you.

Note that if I `write_rgb` instead of trying to manually compose the YUV420 byte array, I manage to send the output as `BGR3`.

**Comment by ralfbiedert** [comment]
Maybe I misunderstand, but isn't that the stride you're looking at? What does `strides_yuv()` say?

**Comment by tgmichel** [comment]
> Maybe I misunderstand, but isn't that the stride you're looking at? What does strides_yuv() say?

Yeah my bad, this is my first time working with video encoders. `strides_yuv() `outs `(1984, 992, 992)`
